packages/now-build-utils/src/detect-routes.ts

```diff
@@ -316,14 +299,35 @@ export async function detectRoutes(
   builders: Builder[],
   featHandleMiss = false
 ): Promise<RoutesResult> {
-  const routesResult = await detectApiRoutes(files, builders, featHandleMiss);
+  const routesResult = await detectApiRoutes(files, builders);
+  const { defaultRoutes, isDynamic } = routesResult;
   const directory = detectOutputDirectory(builders);
-
-  if (routesResult.defaultRoutes && directory && !featHandleMiss) {
-    routesResult.defaultRoutes.push({
-      src: '/(.*)',
-      dest: `/${directory}/$1`,
-    });
+  if (defaultRoutes) {
+    const hasApiRoutes = defaultRoutes.length > 0;
+    if (hasApiRoutes) {
+      defaultRoutes.push({
+        status: 404,
+        src: '/api(/.*)?$',
+      });
+    }
+    if (directory) {
+      defaultRoutes.push({
+        src: '/(.*)',
+        dest: `/${directory}/$1`,
```

Comment:
We shouldn't need this

Reply:
At this time, I'm not renaming output files so the routes will need to remain here.